### PR TITLE
remove duplicated validation from podsecuritypolicy

### PR DESCRIPTION
**Pull request by CaoShuFeng**
**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: pkg/security/podsecuritypolicy/provider.go [hunk 1]
```diff
@@ -273,10 +273,6 @@ func (s *simpleProvider) ValidateContainerSecurityContext(pod *api.Pod, containe
 
 	allErrs = append(allErrs, s.strategies.CapabilitiesStrategy.Validate(pod, container, sc.Capabilities())...)
 
-	if !s.psp.Spec.HostNetwork && podSC.HostNetwork() {
```
Comment (CaoShuFeng) [inline pkg/security/podsecuritypolicy/provider.go]: These three has been checked in line 196.

File: plugin/pkg/admission/security/podsecuritypolicy/admission_test.go [hunk 2]
```diff
@@ -2205,14 +2205,11 @@ func TestPolicyAuthorizationErrors(t *testing.T) {
...
 		t.Run(desc, func(t *testing.T) {
-			var (
-				authz      = &TestAuthorizer{usernameToNamespaceToAllowedPSPs: tc.allowed}
-				privileged = true
-			)
+			authz := &TestAuthorizer{usernameToNamespaceToAllowedPSPs: tc.allowed}
 			pod := goodPod()
 			pod.Namespace = ns
 			pod.Spec.ServiceAccountName = sa
-			pod.Spec.Containers[0].SecurityContext.Privileged = &privileged
```
Comment (liggitt) [inline plugin/pkg/admission/security/podsecuritypolicy/admission_test.go]: why was this change needed?

Comment (CaoShuFeng) [reply plugin/pkg/admission/security/podsecuritypolicy/admission_test.go]: This change is not necessary. Test still passes without this change.

The motivation of this change:
It ensures that `HostPID` will only cause one validationErr for each policy, not two.

https://github.com/kubernetes/kubernetes/pull/56848/files#diff-817778329c5e0a52edffd71d550c4692R2220

**Comment by deads2k** [comment]
@tallclair it's been a month.  Still good?
